proof/refine/ARM/Schedule_R.thy

```diff
@@ -1102,8 +1119,12 @@ lemma tcbSchedDequeue_invs_no_cicd'[wp]:
   done
 
 lemma switchToThread_invs_no_cicd':
-  "\<lbrace>invs_no_cicd' and st_tcb_at' runnable' t and tcb_in_cur_domain' t \<rbrace> ThreadDecls_H.switchToThread t \<lbrace>\<lambda>rv. invs' \<rbrace>"
-  apply (simp add: Thread_H.switchToThread_def)
+  "\<lbrace>invs_no_cicd' and st_tcb_at' runnable' t and tcb_in_cur_domain' t \<rbrace>
+   do  y <- ARM_H.switchToThread t;
+       y <- tcbSchedDequeue t;
+       setCurThread t
+    od
+   \<lbrace>\<lambda>rv. invs' \<rbrace>"
```

Comment:
Could you give a quick explanation for why this is now a lemma about multiple functions? It's not immediately obvious to me, and if nothing else maybe the name should change.

Reply:
Yes! So I added the `ready_qs_runnable` assert to `switchToThread`. However, if I had `switchToThread` in this lemma, then when we get to the lemma `switchToThread_lookupBitmapPriority_wp` below where this lemma is used, I can't generate the fact from the assert because it's hidden beneath the function definition. So I have this lemma here, then in the lemma below, I unfold `switchToThread`, use an sp rule to generate the fact from the assert, then the proof continues as before. 

But you're right that I should at least change the name. I think I'll just tack `_helper` on the end. Let me know if you have another preference though

Reply:
Well, actually, this lemma is used in a few other places, so I'm not sure just calling it "helper" is best. Would a comment beforehand be best, then?

Reply:
Ignore my previous comment; the other times the lemma is mentioned, it's deleted from the wp set. Which is strange because it's not in the wp set. Hmm. I've decided to delete these lines (though maybe if they're useful because they indicate that we certainly do not want that lemma being used in wp, I can add them back). I've decided to just go with `switchToThread_invs'_helper`

Reply:
They're likely removed there because it was in the set at some previous time until someone realised that was not a good idea, but did not clean up the other mentions.